tests/test_grammar.py

```diff
@@ -1,304 +1,437 @@
 from __future__ import absolute_import
+import re
 
 import os
-from unittest import TestCase, main
+from unittest import TestCase, main, SkipTest
 
 from lark import Lark, Token, Tree, ParseError, UnexpectedInput
 from lark.load_grammar import GrammarError, GRAMMAR_ERRORS, find_grammar_errors, list_grammar_imports
 from lark.load_grammar import FromPackageLoader
+from lark.lark_validator import LarkValidator
+
+
+__all__ = []
+
+class LarkDotLark:
+    def __init__(self, grammar, **kwargs):
+        options = {}
+        options.update(kwargs)
+        if "start" in options and options["start"] != "start":
+            # We're not going to parse with the parser, so just override it.
+            options["start"] = "start"
+        options["propagate_positions"] = True
+        lark_parser = Lark.open_from_package("lark", "grammars/lark.lark", **options)
+        tree = lark_parser.parse(grammar)
+        LarkValidator.validate(tree)
+
+    def parse(self, text: str, start=None, on_error=None):
+        raise SkipTest("Cannot test cases with lark.lark that try to parse using the tested grammar.")
+
+
+def _make_tests(parser):
+    class _TestGrammar(TestCase):
+        def test_errors(self):
+            # TODO test_errors needs a lot of work for lark.lark.
+            if parser == LarkDotLark:
+                self.skipTest("test_errors needs a lot of work for lark.lark.")
+            for msg, examples in GRAMMAR_ERRORS:
+                for example in examples:
+                    with self.subTest(example=example):
+                        self.assertRaisesRegex(GrammarError, re.escape(msg), parser, example)
+
+        def test_empty_literal(self):
+            # Issues #888
+            self.assertRaisesRegex(GrammarError, "Empty terminals are not allowed \(\"\"\)", parser, "start: \"\"")
+
+        def test_ignore_name(self):
+            spaces = []
+            p = parser("""
+                start: "a" "b"
+                WS: " "
+                %ignore WS
+            """, parser='lalr', lexer_callbacks={'WS': spaces.append})
+            assert p.parse("a b") == p.parse("a    b")
+            assert len(spaces) == 5
+
+        def test_override_rule1(self):
+            # Overrides the 'sep' template in existing grammar to add an optional terminating delimiter
+            # Thus extending it beyond its original capacity
+            self.skipTest("Test fails for lark.lark because it doesn't execute %import.")
+            p = parser("""
+                %import .test_templates_import (start, sep)
+
+                %override sep{item, delim}: item (delim item)* delim?
+                %ignore " "
+            """, source_path=__file__)
+
+            a = p.parse('[1, 2, 3]')
+            b = p.parse('[1, 2, 3, ]')
+            assert a == b
+
+        def test_override_rule2(self):
+            if parser == LarkDotLark:
```

Comment:
If you always skip for LarkDotLark, why not take it out of `_make_tests`?

Same question applies to the other tests with the same pattern.

Reply:
It seemed easier to not break the file up that way.  If you want me to do that, I will.

Reply:
@erezsh Would you like me to do that?

Reply:
If I said so, then I guess so, yes :)

Unless there is a reason not to.